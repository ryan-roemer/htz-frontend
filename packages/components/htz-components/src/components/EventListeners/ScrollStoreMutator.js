import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import { graphql, } from 'react-apollo';
import gql from 'graphql-tag';

export const UPDATE_SCROLL = gql`
  mutation updateScroll($x: Int!, $y: Int!, $direction: String!, $velocity: Int!) {
    updateScroll(x: $x, y: $y, direction: $direction, velocity: $velocity) @client {
      scroll {
        x
        y
        velocity
      }
    }
  }
`;

const propTypes = {
  /**
   *  The Throttling used by `<ScrollListener/>` on the scroll event listener, needed to calculate the velocity.
   */
  throttle: PropTypes.number.isRequired,
  /**
   * The x scroll position
   */
  x: PropTypes.number.isRequired,
  /**
   * The y scroll position
   */
  y: PropTypes.number.isRequired,
  /**
   * A function that calls the Mutation that updates the Apollo store.
   *
   * Passed implicitly by Apollo, not directly as an attribute on the component.
   */
  mutate: PropTypes.func.isRequired,
};

export class ScrollStoreMutator extends Component {
  componentWillReceiveProps(nextProps) {
    const { x, y, mutate, throttle, } = this.props;
    const velocity = (y - nextProps.y) / throttle;
    if (!(y === nextProps.y && x === nextProps.x)) {
      mutate({ variables: { x: nextProps.x, y: nextProps.y, velocity, }, });
    }
  }
  render() {
    return null;
  }
}

ScrollStoreMutator.propTypes = propTypes;

export default graphql(UPDATE_SCROLL)(ScrollStoreMutator);
