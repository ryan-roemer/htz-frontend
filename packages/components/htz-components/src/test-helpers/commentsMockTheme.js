const mockTheme = {
  // Constants
  bps: Object.freeze({
    widths: Object.freeze({
      s: 600,
      m: 768,
      l: 1024,
      xl: 1280,
    }),
    misc: Object.freeze({
      landscape: '(orientation: landscape)',
      portrait: '(orientation: portrait)',
      hidpi: '(min-resolution: 1.5dppx)',
    }),
  }),
  btnStyle: Object.freeze({
    // Border width
    borderBottomWidth: 1,
    borderEndWidth: 1,
    borderStartWidth: 1,
    borderTopWidth: 1,

    // Border style
    borderBottomStyle: 'solid',
    borderEndStyle: 'solid',
    borderStartStyle: 'solid',
    borderTopStyle: 'solid',
    radius: 0,
    boxModel: { hp: 4, vp: 1, },

    // Font Style
    fontWeight: 'bold',
  }),
  commentsStyle: Object.freeze({
    textInputVariant: 'primary',
  }),
  inputStyle: Object.freeze({
    borderWidth: 1,
    lines: 1,
    borderStyle: 'solid',
    radius: 0,
    height: 6,
    // typographic Scale from theme
    typeScale: -1,
    fontWeightLabel: 'bold',
  }),
  selectStyle: Object.freeze({
    borderWidth: 1,
    lines: 0,
    borderStyle: 'solid',
  }),
  typeConf: Object.freeze({
    default: Object.freeze({
      base: 16,
      minPadding: 2,
      ratio: 2,
      rhythmUnit: 6,
      steps: 5,
    }),
    xl: Object.freeze({
      base: 18,
      minPadding: 2,
      ratio: 2,
      rhythmUnit: 7,
      steps: 5,
    }),
  }),

  commentI18n: Object.freeze({
    tags: Object.freeze({
      editorsPick: 'בחירת העורכים',
      usersPick: 'בחירת הגולשים',
    }),
    buttons: Object.freeze({
      readMoreBtnTxt: 'קרא עוד',
      replyBtnTxt: 'הגב',
      reportAbuseBtnTxt: 'דווח כפוגעני',
    }),
  }),

  commentFormI18n: Object.freeze({
    buttons: Object.freeze({
      sendBtnTxt: 'שלח',
      cancelBtnTxt: 'בטל',
      toggleUserBtnText: 'להוספת תגובה מזוהה לחץ כאן',
    }),
    labels: Object.freeze({
      nameLabelTxt: 'שם',
      commentLabelTxt: 'תגובה',
    }),
    notes: Object.freeze({
      nameNoteTxt: 'הזינו שם שיוצג כמחבר התגובה',
      commentNoteTxt: 'בשליחת תגובה זו הנני מצהיר שהינני מסכים/ה עם תנאי השימוש של אתר הארץ',
    }),
    errorNotes: Object.freeze({
      nameErrorNoteTxt: 'חובה להזין שם',
      commentErrorNoteTxt: 'נא להזין את תוכן התגובה',
    }),
  }),
  commentSentI18n: Object.freeze({
    buttons: Object.freeze({
      getNotificationsBtnTxt: 'עדכנו אותי',
      dontGetNotificationsBtnTxt: 'לא תודה',
      closeBtnText: 'סגור',
    }),
    labels: Object.freeze({
      emailLabelTxt: 'דוא"ל',
    }),
    notes: Object.freeze({
      emailNoteTxt: 'הזינו כתובת דוא"ל לקבלת התראות',
    }),
    errorNotes: Object.freeze({
      emailErrorNoteTxt: 'נא להזין כתובת דוא"ל תקינה',
    }),
    texts: Object.freeze({
      commentRecievedBoldText: 'תגובתך נקלטה בהצלחה, ',
      commentRecievedText: 'ותפורסם על פי מדיניות המערכת.',
      commentRecievedTextSecondRow: 'באפשרותך לקבל התראה בדוא"ל כאשר תגובתך תאושר ותפורסם',
      commentRecievedBoldTextThankYouPage: 'תודה!',
      commentRecievedTextThankYouPage: 'תגובתך נקלטה בהצלחה ותפורסם על פי מדיניות המערכת',
    }),
  }),
  commentsSectionI18n: Object.freeze({
    buttons: Object.freeze({
      loadAllCommentsBtnText: 'טען את כל התגובות',
    }),
    selectItems: Object.freeze({
      dateDescendingItemTxt: 'מהאחרונה לראשונה',
      dateAscendingItemTxt: 'מהראשונה לאחרונה',
      commentRatingItemTxt: 'הצג לפי דירוג',
      editorsPickItemTxt: 'בחירת העורכים',
    }),
    texts: Object.freeze({
      chooseSortMethodText: 'סדרו את התגובות',
    }),
  }),

  // Methods
  color: () => 'red',
  getTransition: () => ({
    transitionDuration: undefined,
    transitionTimingFunction: undefined,
    transitionDelay: undefined,
  }),
  mq: () => () => ({
    '@media (min-width: 37.5em) and (max-width: 64em)': {
      color: 'red',
    },
  }),
  type: () => () => ({}),
};

export default mockTheme;
