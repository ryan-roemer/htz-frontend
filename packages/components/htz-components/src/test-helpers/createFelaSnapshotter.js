import React from 'react';
import renderer from 'react-test-renderer';
// eslint-disable-next-line import/no-extraneous-dependencies
import { renderToString, } from 'fela-tools';

// function Wrapper({children,}) {
//   return children;
// }
export default function createFelaSnapshotter(StyleProvider, testRenderer) {
  return function testFelaSnapshot(component) {
    const testObj = {
      // Output an object with both the component snapshot and raw styles.
      component: renderer
        .create(
          <StyleProvider renderer={testRenderer}>
            <div>{component}</div>
          </StyleProvider>
        )
        .toJSON(),
      styles: renderToString(testRenderer),
    };

    return testObj;
  };
}
