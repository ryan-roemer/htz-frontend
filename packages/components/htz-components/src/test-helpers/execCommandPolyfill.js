/* global document */
/**
 * curruntly jest doesnt support document.execCommand
 * the following function act as a polyfill's for this function, it returns the prev
 * function so we can cancel the polyfill at the end of the test
 */
export default function execCommandPolyfill() {
  const prevExecCommand = document.execCommand;
  document.execCommand = document.execCommand || function noop() {};
  return prevExecCommand;
}
