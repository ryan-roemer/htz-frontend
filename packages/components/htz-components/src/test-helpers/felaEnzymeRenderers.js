// eslint-disable-next-line import/no-extraneous-dependencies
import htzTheme from '@haaretz/htz-theme';
import { createRenderer, createFelaEnzymeRenderers, } from '@haaretz/fela-utils';
// import createFelaEnzymeRenderers from '../utils/createFelaEnzymeRenderers';

const testRenderer = createRenderer({ isRtl: true, });
const { felaMount, felaShallow, } = createFelaEnzymeRenderers(
  testRenderer,
  htzTheme
);

export { felaMount, felaShallow, };
