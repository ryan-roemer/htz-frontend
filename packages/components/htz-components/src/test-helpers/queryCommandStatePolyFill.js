/* global document */

/**
 * curruntly jest doesnt support document.queryCommandState()
 * the following function act as a polyfill's for this function, it returns the prev
 * function so we can cancel the polyfill at the end of the test
 */
export default function queryCommandStatePolyFill() {
  const prevQueryCommandState = document.queryCommandState;
  document.queryCommandState = document.queryCommandState || (() => true);

  return prevQueryCommandState;
}
