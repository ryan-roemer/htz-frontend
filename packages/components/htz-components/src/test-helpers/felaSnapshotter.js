import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
