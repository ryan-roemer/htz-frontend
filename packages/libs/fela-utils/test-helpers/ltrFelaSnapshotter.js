import createRenderer from '../src/createRenderer';
import StyleProvider from '../src/StyleProvider';
import createFelaSnapshotter from '../src/createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: false, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer, {});

export default felaSnapshotter;
