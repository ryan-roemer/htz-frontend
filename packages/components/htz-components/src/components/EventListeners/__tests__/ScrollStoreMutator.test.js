import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';
import { felaMount, } from '../../../test-helpers/felaEnzymeRenderers';
import { ScrollStoreMutator, UPDATE_SCROLL, } from '../ScrollStoreMutator';

describe('<ScrollStoreMutator>', () => {
  describe('DOM element', () => {
    it('correctly doesnt render anything with minimal required props', () => {
      const mockFunc = jest.fn();
      const { component, styles, } = felaSnapshotter(
        <ScrollStoreMutator throttle={100} x={0} y={0} mutate={mockFunc} />
      );
      expect(component).toMatchSnapshot();
      expect(styles).toMatchSnapshot();
    });
    it('calls mutate function with the correct params', () => {
      const onMutate = jest.fn();
      const output = felaMount(<ScrollStoreMutator throttle={100} x={0} y={0} mutate={onMutate} />);
      output.setProps({ y: 100, x: 50, });
      expect(onMutate).toHaveBeenCalledWith({ variables: { x: 50, y: 100, velocity: -1, }, });
    });
    it('calls doesnt call mutate if x and y dont change', () => {
      const onMutate = jest.fn();
      const output = felaMount(
        <ScrollStoreMutator throttle={100} x={100} y={50} mutate={onMutate} />
      );
      output.setProps({ y: 50, x: 100, });
      expect(onMutate).toHaveBeenCalledTimes(0);
    });
    it('has the same mutation query string', () => {
      expect(UPDATE_SCROLL).toMatchSnapshot();
    });
  });
});
