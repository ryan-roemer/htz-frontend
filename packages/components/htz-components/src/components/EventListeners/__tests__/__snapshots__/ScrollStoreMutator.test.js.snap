// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollStoreMutator> DOM element correctly doesnt render anything with minimal required props 1`] = `<div />`;

exports[`<ScrollStoreMutator> DOM element correctly doesnt render anything with minimal required props 2`] = `""`;

exports[`<ScrollStoreMutator> DOM element has the same mutation query string 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "updateScroll",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "x",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "x",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "y",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "y",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "direction",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "direction",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "velocity",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "velocity",
                  },
                },
              },
            ],
            "directives": Array [
              Object {
                "arguments": Array [],
                "kind": "Directive",
                "name": Object {
                  "kind": "Name",
                  "value": "client",
                },
              },
            ],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "updateScroll",
            },
            "selectionSet": undefined,
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "x",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "y",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "direction",
            },
          },
        },
        Object {
          "defaultValue": undefined,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "velocity",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 195,
    "start": 0,
  },
}
`;
